travharv/web_discovery.py

```diff
@@ -90,31 +90,24 @@ def get_description_into_graph(
     session.mount("http://", adapter)
     session.mount("https://", adapter)
 
-    formats = ["text/turtle", "application/ld+json"]
-    ACCEPTABLE_MIMETYPES = {
-        "application/ld+json",
-        "text/turtle",
-        "application/json",
-    }
     triples_found = False
 
-    for format in formats:
-        headers = {"Accept": format}
-        log.debug(f"requesting {subject_url} with {headers=}")
-        r = session.get(subject_url, headers=headers)
-        mime_type, options = cgi.parse_header(r.headers["Content-Type"])
-
-        if r.status_code == 200 and bool(mime_type in ACCEPTABLE_MIMETYPES):
-            triples_found = True
-            try:
-                graph.parse(data=r.text, format=format, publicID=subject_url)
-                log.info(
-                    f"content of{subject_url} added to triplestore in{format=}"
-                )
-            except Exception as e:
-                log.warning(
-                    f"failed to parse {subject_url} in {format=} error: {e}"
-                )
+    headers = {"Accept": format}
+    log.debug(f"requesting {subject_url} with {headers=}")
+    r = session.get(subject_url, headers=headers)
+    mime_type, options = cgi.parse_header(r.headers["Content-Type"])
+
+    if r.status_code == 200 and bool(mime_type == format):
```

Comment:
too restrictive

why would you ignore another format then the one you ask for?  this is the whole idea of content-negotiation

sure you can ask for jsonld, but if you get ttl use that -- reintroduce ACCEPTIBLE_MIMETYPES and keep anything that fits that

the fact that an end-user would want you to actively both get type A and B can lead to you in both cases receiving the same B format and using that, no problem

reversely however, if I know ask for B and only A is offered: this code drops it and you end up with nothing

Reply:
Ok will reintroduce the mimetypes again in a next push